samples/GenericWebHost/WebHostExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.ObjectPool;
+
+namespace GenericWebHost
+{
+    public static class WebHostExtensions
+    {
+        public static IHostBuilder ConfigureWebHost(this IHostBuilder builder, Action<HostBuilderContext, IApplicationBuilder> configureApp)
+        {
+            return builder.ConfigureServices((bulderContext, services) =>
+            {
+                services.Configure<WebHostServiceOptions>(options =>
+                {
+                    options.ConfigureApp = configureApp;
```

Comment:
Options hmm.

Reply:
Another option would be to use `IStartup` (a modified version with only a `Configure` method) and have this register a delegate-based Startup class.

Reply:
We'll iterate on the generic web host design between now and 3.0. We're starting with the proof of concept.

Reply:
Agreed